index.ts

```diff
@@ -1,17 +1,55 @@
 type SmartlookWindow = Window & { smartlook?: any }
 const SL_NOT_INITIALIZED = 'Smartlook client is not initialized.'
 
+interface Network {
+	allowedHeaders?: string[]
+	allowedUrls?: (string | RegExp)[]
+}
+
+type RequestOrResponse = {
+	body?: string
+	headers?: Record<string, string[]>
+}
+
+interface Interceptors {
+	network?: (
+		data: {
+			request?: RequestOrResponse
+			response?: RequestOrResponse
+			url: string
+		},
+		context: { pageUrl: string },
+	) => void | false
+	url?: (data: { url: string }) => void
+}
+
+type InitParams = {
```

Comment:
this and maybe above interface as well coudl get exported so customer can use it like this:

```ts
const interceptors: Interceptors = ...
const config: InitParams = {..., interceptors}
smartlook.init('key', config)
```

what do you think @smartlook/web-sdk?